projects/plugins/jetpack/_inc/client/main.jsx

```diff
@@ -576,9 +576,7 @@ export default connect(
 	withDispatch( dispatch => {
 		return {
 			setConnectionStatus: connectionStatus => {
-				dispatch( CONNECTION_STORE_ID ).startResolution( 'getConnectionStatus', [] );
 				dispatch( CONNECTION_STORE_ID ).setConnectionStatus( connectionStatus );
```

Comment:
Reading the comments [here](https://unfoldingneurons.com/2020/wordpress-data-store-properties-resolvers) I think that if we already have the value in state, the resolver shouldn't even be hit when the selector is called.

I think that if we insist in calling `setConnectionStatus` here we still need to call start/finishResolution in order to avoid the remote API call.

How we handle initialState is different from setting the `connectionStatus` to a custom object passed by another consumer, such as the Jetpack plugin.

However given we want to use the InitialState here, we might want to consider dropping this approach and just fetch the connectionStatus the same way we do for the Backup plugin, aka via the corresponding selector:

```
const connectionStatus = useSelect(
		select => select( CONNECTION_STORE_ID ).getConnectionStatus(),
		[]
	);
```

Does that make sense? cc @sergeymitr

Reply:
@fgiannar you are right. The API call will still be triggered. That's why I added additional instructions to the testing.

We either need to go with an approach suggested here: https://github.com/Automattic/jetpack/pull/21721

Or we can rely always on the initial state and remove that selector from the store...